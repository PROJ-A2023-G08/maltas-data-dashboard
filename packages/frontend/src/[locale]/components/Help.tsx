import React, { useState } from "react";
import FaqTab from "./FaqTab";
import ContactTab from "./ContactTab";

const Help = () => {
  const [activeTab, setActiveTab] = useState("faq");

  const toggleTab = (tab: any) => {
    setActiveTab(tab);
  };

  return (
    <div>
      <div className="flex justify-center mb-4 mt-20">
        <button
          onClick={() => toggleTab("faq")}
          className={`mr-2 pt-2 pb-2 pl-20 pr-20 ${
            activeTab === "faq" ? "bg-black text-white" : "bg-white"
          } rounded-md hover:bg-gray-600 focus:outline-none`}
        >
          FAQ
        </button>
        <button
          onClick={() => toggleTab("contact")}
          className={`mr-2 pt-2 pb-2 pl-20 pr-20 ${
            activeTab === "contact" ? "bg-black text-white" : "bg-white"
          } rounded-md hover:bg-gray-600 focus:outline-none`}
        >
          Contact
        </button>
      </div>
      {activeTab === "faq" ? <FaqTab /> : null}
      {activeTab === "contact" ? <ContactTab /> : null}
    </div>
  );
};

export default Help;
