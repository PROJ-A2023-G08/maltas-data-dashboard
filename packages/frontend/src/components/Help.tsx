import React, { useState } from "react";
import FaqTab from "./FaqTab";
import ContactTab from "./ContactTab";
import { useTranslation } from "next-i18next";

const Help = () => {
  const { t } = useTranslation("common");
  const [activeTab, setActiveTab] = useState("faq");

  const toggleTab = (tab: any) => {
    setActiveTab(tab);
  };

  return (
    <div>
      <div className="flex justify-center mb-4 mt-20">
        <button
          onClick={() => toggleTab("faq")}
          className={`mr-2 pt-2 pb-2 pl-20 pr-20 ${
            activeTab === "faq" ? "bg-black text-white" : "bg-white"
          } rounded-md hover:bg-gray-600 focus:outline-none`}
        >
          {t("uiElements.FAQ")}
        </button>
        <button
          onClick={() => toggleTab("contact")}
          className={`mr-2 pt-2 pb-2 pl-20 pr-20 ${
            activeTab === "contact" ? "bg-black text-white" : "bg-white"
          } rounded-md hover:bg-gray-600 focus:outline-none`}
        >
          {t("uiElements.Contact")}
        </button>
      </div>
      {activeTab === "faq" ? <FaqTab /> : null}
      {activeTab === "contact" ? <ContactTab /> : null}
    </div>
  );
};

export default Help;
