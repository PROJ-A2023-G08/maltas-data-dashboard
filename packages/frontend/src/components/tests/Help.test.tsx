import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Help from "../Help";
import "@testing-library/jest-dom";

jest.mock("next-i18next", () => ({
  useTranslation: jest.fn().mockReturnValue({
    // @ts-ignore
    t: (str) => str,
  }),
}));

it("Help component loads", () => {
  const setActiveComponent = jest.fn();
  const tree = renderer.create(<Help />).toJSON();
  expect(tree).toMatchSnapshot();
});
