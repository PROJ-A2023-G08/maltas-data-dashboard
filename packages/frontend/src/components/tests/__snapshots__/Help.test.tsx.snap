// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help component loads 1`] = `
<div>
  <div
    className="flex justify-center -mb-12 mt-40"
  >
    <button
      className="mr-2 pt-2 pb-2 pl-20 pr-20 bg-black text-white  rounded-md hover:bg-gray-600 focus:outline-none"
      onClick={[Function]}
    >
      uiElements.FAQ
    </button>
    <button
      className="mr-2 pt-2 pb-2 pl-20 pr-20 bg-white rounded-md hover:bg-gray-600 focus:outline-none"
      onClick={[Function]}
    >
      uiElements.Contact
    </button>
  </div>
  <div
    className="container mx-auto p-4 w-1/2 pt-20"
  >
    <h1
      className="text-4xl font-bold mb-4 text-start ml-20"
    >
      FAQ
    </h1>
    <ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
      <ul
        className="mb-4"
      >
        <button
          className="flex items-center justify-between w-full bg-white p-2 rounded-md border border-gray-300 focus:outline-none"
          onClick={[Function]}
        >
          <span
            className="text-lg font-bold"
          >
            What is Lorem ipsum
          </span>
          <span
            className="ml-2"
          >
            ▼
          </span>
        </button>
      </ul>
    </ul>
  </div>
</div>
`;
